packages/generator/ni_measurement_plugin_sdk_generator/client/_support.py

```diff
@@ -0,0 +1,173 @@
+        configuration_metadata,
+        metadata.measurement_signature.configuration_defaults.value,
+        service_class + ".Configurations",
+    )
+
+    default_values = [None] * len(deserialized_parameters)
+    for k, v in deserialized_parameters.items():
+        configuration_metadata[k] = configuration_metadata[k]._replace(default_value=v)
+        default_values[k - 1] = v
+
+    return configuration_metadata
+
+
+def _get_output_metadata_by_index(
+    metadata: v2_measurement_service_pb2.GetMetadataResponse,
+) -> Dict[int, ParameterMetadata]:
+    output_parameter_list = _get_output_parameters(metadata)
+    output_metadata = frame_metadata_dict(output_parameter_list)
+    return output_metadata
+
+
+def _remove_invalid_characters(input_string: str, new_char: str) -> str:
+    # Replace any spaces or special characters with an '_'.
+    if not input_string.isidentifier():
+        for invalid_char in INVALID_CHARS:
+            input_string = input_string.replace(invalid_char, new_char)
+
+    if input_string[0].isdigit() or keyword.iskeyword(input_string):
+        input_string = "_" + input_string
+
+    return input_string
+
+
+def _get_python_module_name(input_string: str) -> str:
+    module_name = input_string.replace(" ", "_").lower()
+    module_name = _remove_invalid_characters(module_name, "")
+    return module_name
+
+
+def _get_python_identifier(input_string: str) -> str:
+    valid_identifier = input_string.lower()
+    valid_identifier = _remove_invalid_characters(valid_identifier, "_")
+    return valid_identifier
+
+
+def _get_python_type_as_str(type: Field.Kind.ValueType, is_array: bool) -> str:
+    python_type = PROTO_DATATYPE_TO_PYTYPE_LOOKUP.get(type)
+
+    if python_type is None:
+        raise TypeError("Invalid data type: The configured data type is unsupported.")
+
+    if is_array:
+        return f"List[{python_type.__name__}]"
+    return python_type.__name__
+
+
+def _get_configuration_parameters_with_type_and_default_values(
+    configuration_metadata: Dict[int, ParameterMetadata],
+    import_modules: Dict[str, str],
+) -> Tuple[str, str]:
+    configuration_parameters = []
+    parameter_names = []
+
+    for metadata in configuration_metadata.values():
+        parameter_name = _get_python_identifier(metadata.display_name)
+        parameter_names.append(parameter_name)
+
+        default_value = metadata.default_value
+        parameter_type = _get_python_type_as_str(metadata.type, metadata.repeated)
+        if isinstance(default_value, str):
+            default_value = f'"{default_value}"'
+
+            # If it's path type, make the value as raw string literal to ignore escape characters.
+            if metadata.annotations and metadata.annotations["ni/type_specialization"] == "path":
+                default_value = f"r{default_value}"
+                parameter_type = "Path"
+                import_modules[PATH_IMPORT] = ImportType.Built_In.name
```

Comment:
ImportType.BUILT_IN was not working in `template` file, so, I went with string. Now updated with `type(import_type).BUILT_IN` in `template` file  and updated the `import_modules: Dict[str, ImportType]`.

Reply:
`ImportType.BUILT_IN` should work in any Python module or Mako template you want, but you have to import `ImportType` like any other name.

`type(import_type).BUILT_IN` is a roundabout solution compared to importing the type or the module containing it.

You have since replaced the dict with two sets, which seems more reasonable, aside from the ordering problem it introduces (flagged separately).